src/OrbitGl/AnnotationTrack.h

```diff
@@ -24,6 +24,9 @@ class AnnotationTrack {
   [[nodiscard]] std::optional<std::pair<std::string, double>> GetValueLowerBound() const {
     return value_lower_bound_;
   }
+  [[nodiscard]] std::string GetValueUpperBoundTooltips() const {
+    return value_upper_bound_tooltips_;
+  }
```

Comment:
I would do this differently. You are using inheritance for a lot of things, why not here? You could do:
```
[[nodiscard]] virtual std::string GetValueUpperBoundTooltip() const { return ""; }
```

(Also btw: The method suffix here is `Tooltips`, just `Tooltip` (singular) is better)

And then in CGroupAndProcessMemoryTrack.h
`std::string GetValueUpperBoundTooltip() const override;`

CGroupAndProcessMemoryTrack.cpp
```
std::string CGroupAndProcessMemoryTrack::GetValueUpperBoundTooltip() const {
  const std::string kGameCGroupName = "game";
  constexpr float kGameCGroupLimitGB = 7;
  if (cgroup_name_ == kGameCGroupName) {
    return absl::StrFormat(
        "<b>The memory production limit of the %s cgroup is %.2fGB</b>.<br/><br/>"
        "<i>To launch game with the production cgroup limits, add the flag "
        "'--enforce-production-ram' to the 'ggp run' command</i>.",
        kGameCGroupName, kGameCGroupLimitGB);
  }
}
```

And in AnnotationTrack.cpp
instead of `if (!value_upper_bound_tooltips_.empty())` it can be `if (!GetValueUpperBoundTooltip().empty())`

Reply:
Good idea! Changed accordingly. Thanks!